pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -114,15 +138,16 @@ func doSetup(sourceURL url.URL, usernameSecret, passwordSecret, tokenSecret stri
 		username = DefaultUsername
 	}
 
-	// Remove user/pw from the source url
-	overrideSourceURL := sourceURL
-	overrideSourceURL.User = nil
+	overrideSourceURL, err := overrideURLFn(&sourceURL, username, password)
```

Comment:
this wasn't quite what I meant... we need to be able to unit test the proxy conditions as well... I would call the `overrideURLFn` `startProxy`, and move the check for password length out of it... that way, the unit test can pass in a stub, and ensure it only gets called when needed